service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -278,4 +298,27 @@ void getDatasetPreviewTable() {
         datasetPreviewTable.getColumns().get(0),
         is(new bio.terra.catalog.model.ColumnModel().name("column a").arrayOf(false)));
   }
+
+  @Test
+  void getDatasetPreviewTableWorkspace() {
+    var tdrDataset =
+        new Dataset(dataset.id(), sourceId, StorageSystem.TERRA_WORKSPACE, metadata, null);
+    var tableName = "sample";
+    List<String> att = List.of("a", "b", "c");
+    EntityTypeMetadata ent = new EntityTypeMetadata().idName("idName").attributeNames(att);
+    Map<String, EntityTypeMetadata> map = new HashMap<>();
+    map.put("str", new EntityTypeMetadata());
+    map.put(tableName, ent);
+    when(datasetDao.retrieve(datasetId)).thenReturn(tdrDataset);
+    when(rawlsService.entityMetadata(user, tdrDataset.storageSourceId())).thenReturn(map);
+    when(rawlsService.entityQuery(user, tdrDataset.storageSourceId(), tableName))
+        .thenReturn(new EntityQueryResponse().results(List.of()));
+    DatasetPreviewTable datasetPreviewTable =
+        datasetService.getDatasetPreview(user, tdrDataset.id(), tableName);
+    assertThat(datasetPreviewTable.getRows(), empty());
+    assertThat(datasetPreviewTable.getColumns(), hasSize(4));
+    assertThat(
+        datasetPreviewTable.getColumns().get(0),
+        is(new bio.terra.catalog.model.ColumnModel().name("idName")));
```

Comment:
RawlsService does not return the data type for the columns. Our api returns whether or not the column is an array. Is this even something we want to implement?

Reply:
I think it's OK. We may want to revisit this and remove the type information from our APIs, if the UI doesn't need this information to render the preview.